src/components/MasterItemsSelect.js

```diff
@@ -6,8 +6,10 @@ class MasterItemsSelect extends Component {
     super(props);
     this.handleChange = this.handleChange.bind(this);
     this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    const filteredItems = this.filterItems(this.props.masterItems[0].category);
     this.state = {
-      filteredItems: this.filterItems(this.props.masterItems[0].category)
+      filteredItems,
+      selectedItemId: filteredItems[0].id
```

Comment:
feels kind of weird...had to track the selectedItemId so the display value on the items dropdown is in sync with the selectedItem passed to props.onSelectMasterItem.